### PR TITLE
automation: Use reusable workflows to avoid duplications

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

This PR introduces GitHub reusable workflows to reduce duplicated code. As GitHub doesn't support placing reusable workflows everywhere, I placed them in `workflows` folder (the only place where they can be inside the repo) with the prefix `rw` to distinguish them from regular workflows. If we want to move to another place, we can move them into a shared repo in the org (I don't like it)

The summary of the changes is:
- e2e tests on main branch have been moved to a reusable workflow and replace it in `main-build.yml` and `nightly-e2e.yml`
- ARM64 smoke tests on main branch have been moved to a reusable workflow and replace it in `main-build.yml` and `nightly-e2e.yml`
- trivy scan has been moved to a reusable workflow and replace it in `main-build.yml`, `nightly-e2e.yml` and `pr-validation.yml`
- Smoke tests are executed on `ubuntu-latest` and every supported k8s version.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes https://github.com/kedacore/keda/issues/2569

### CODE REVIEW

**Comment by JorTurFer** [comment]
Important @kedacore/keda-core-contributors , **DON'T MERGE this PR**, I have to modify this line https://github.com/kedacore/keda/pull/3593/files#diff-04af3c9c96028eebf4fa93d5c67e1fa08255fb7a3682f912ccd5bea1dff5ccd4R213 to point `main` instead of `reusable-workflows` branch (GH magic...) 
I have kept `reusalbe-workflows` to be able to run every CI check on PR but before merging I need to change it or next PRs will fail on it

**Review by zroubalik — APPROVED** [review]
Great stuff!

**Comment by JorTurFer** [comment]
[kind team will do a release soon](https://github.com/kubernetes-sigs/kind/issues/2899), so I'll wait to include the node v1.25 too

File: .github/workflows/rw-versions-smoke-tests.yml
Comment (tomkerkhove) [inline .github/workflows/rw-versions-smoke-tests.yml:1]: nit - Instead of using `rw` as prefix, can we move them to `reusable/` or `templates/`? It took me a while to figure out what it meant.

**Review by tomkerkhove — APPROVED** [review]
LGTM, great work! 🥰

File: .github/workflows/rw-versions-smoke-tests.yml
Comment (tomkerkhove) [reply .github/workflows/rw-versions-smoke-tests.yml:1]: 🤦 resolving

Comment (tomkerkhove) [reply .github/workflows/rw-versions-smoke-tests.yml:1]: Can we then at least rename it from `rw` to `template`?

Comment (JorTurFer) [reply .github/workflows/rw-versions-smoke-tests.yml:1]: sure, something like `template-XXXXX`?

Comment (tomkerkhove) [reply .github/workflows/rw-versions-smoke-tests.yml:1]: Yes, please ❤️

Comment (JorTurFer) [reply .github/workflows/rw-versions-smoke-tests.yml]: done :)

**Comment by JorTurFer** [comment]
I need to merge this PR because it's blocking me for other things, once kind releases the new node we can add it in another PR

**Pull request merged**
